### PR TITLE
[cpp-httplib] Update library to 0.5.1

### PR DESCRIPTION
**Pull request by Pospelove**

### CODE REVIEW

File: ports/cpp-httplib/portfile.cmake [hunk 1]
```diff
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (NancyLi1013) [inline ports/cpp-httplib/portfile.cmake]: `include(vcpkg_common_functions)` is no longer needed.
Could you please remove this line?

File: ports/cpp-httplib/portfile.cmake [hunk 2]
```diff
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO yhirose/cpp-httplib
-    REF v0.4.2
-    SHA512 2269bba048790cc37d9dc79de727959d337182ebee50dbacaabcdc495e1a7ef429ad2331c4479b075fd842ba7c3fcef87c487a5c04307e150b747ddd0f04d545
+    REF v0.5.1
+    SHA512 808ad4a9b4d2a6d25eb342985a9d2407c252b6cdf85da2372b7426338c59ccaf49b2a04a4aa1cb0c97487ab8ec6ab5c098e1785edcccd94296488539af6ba1ef
     HEAD_REF master
 )
```
Comment (NancyLi1013) [inline ports/cpp-httplib/portfile.cmake]: Could you update this
```
configure_file(
    ${SOURCE_PATH}/LICENSE
    ${CURRENT_PACKAGES_DIR}/share/cpp-httplib/copyright
    COPYONLY
)
```
as
`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION  ${CURRENT_PACKAGES_DIR}/share/cpp-httplib RENAME copyright)`?

**Comment by NancyLi1013** [comment]
Hi @Pospelove thanks for this PR.
Could you also bump the `Version` in `CONTROL` file?

**Comment by Pospelove** [comment]
Done

**Comment by NancyLi1013** [comment]
It seems that `include(vcpkg_common_functions)` hasn't been removed now.

**Review by NancyLi1013 — APPROVED** [review]

**Pull request merged**